web-wallet/src/routes/(welcome)/setup/restore/__tests__/page.spec.js

```diff
@@ -1,18 +1,295 @@
+import { setKey } from "lamb";
+import { addresses } from "$lib/mock-data";
+import { settingsStore, walletStore } from "$lib/stores";
+import { encryptMnemonic, getSeedFromMnemonic } from "$lib/wallet";
+import loginInfoStorage from "$lib/services/loginInfoStorage";
+import * as walletService from "$lib/services/wallet";
 import Restore from "../+page.svelte";
 
-describe("Restore", () => {
-	afterEach(cleanup);
+/**
+ * @param {HTMLElement} input
+ * @param {*} value
+ * @returns {Promise<boolean>}
+ */
+const fireInput = (input, value) => fireEvent.input(input, { target: { value } });
+
+/** @param {HTMLElement} element */
+function asInput (element) {
+	// eslint-disable-next-line no-extra-parens
+	return /** @type {HTMLInputElement} */ (element);
+}
+
+describe("Restore", async () => {
+	const walletGetPsksSpy = vi.spyOn(Wallet.prototype, "getPsks").mockResolvedValue(addresses);
+	const mnemonic = generateMnemonic();
+	const invalidMnemonic = "dad dad dad dad dad dad dad dad dad dad dad dad";
+	const pwd = "passwordpassword";
+	const loginInfo = await encryptMnemonic(mnemonic, pwd);
+	const seed = getSeedFromMnemonic(mnemonic);
+	const userId = (await new Wallet(seed).getPsks())[0];
+	const getWalletSpy = vi.spyOn(walletService, "getWallet");
+	const gotoSpy = vi.spyOn(appNavigation, "goto");
+	const settingsResetSpy = vi.spyOn(settingsStore, "reset");
+	const originalNavigator = { ...window.navigator };
+	const clearAndInitSpy = vi.spyOn(walletStore, "clearLocalDataAndInit");
+
+	afterEach(async () => {
+		cleanup();
+		clearAndInitSpy.mockClear();
+		getWalletSpy.mockClear();
+		gotoSpy.mockClear();
+		settingsStore.reset();
+		settingsResetSpy.mockClear();
+		walletGetPsksSpy.mockClear();
+		walletStore.reset();
+		Object.defineProperty(window, "navigator", {
+			value: originalNavigator,
+			configurable: true,
+			writable: true
+		});
+	});
+
+	afterAll(() => {
+		clearAndInitSpy.mockRestore();
+		getWalletSpy.mockRestore();
+		gotoSpy.mockRestore();
+		settingsResetSpy.mockRestore();
+		walletGetPsksSpy.mockRestore();
+	});
+
+	it("should render the Terms of Service step of the Restore flow if there is no userId saved in localStorage", () => {
+		const { container } = render(Restore);
+
+		expect(container.firstChild).toMatchSnapshot();
+	});
+
+	it("should render the Existing Wallet notice step of the Restore flow if there is a userId saved in localStorage", () => {
+		settingsStore.update(setKey("userId", userId));
 
-	it("should render the Terms of Service step of the Restore flow", () => {
 		const { container } = render(Restore);
 
 		expect(container.firstChild).toMatchSnapshot();
 	});
+
+	it("should render the Mnemonic Authenticate step after accepting the Existing Wallet Notice and the Terms of Service", async () => {
+		settingsStore.update(setKey("userId", userId));
+
+		const { container, getByRole } = render(Restore);
+
+		const noticeButton = getByRole("button", { name: "Proceed" });
```

Comment:
In general, I don't think you need to store the button in a variable if you don't use it later on. You can just:

```js
await fireEvent.click(getByRole("button", { name: "Proceed" }));
```